crates/sui-framework/sources/display.move

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Defines a Display struct which defines the way an Object
+/// should be displayed. The intention is to keep data as independent
+/// from its display as possible, protecting the development process
+/// and keeping it separate from the ecosystem agreements.
+///
+/// Each of the fields of the Display object should allow for pattern
+/// substitution and filling-in the pieces using the data from the object T.
+module sui::display {
+    use sui::publisher::{is_module, is_package, Publisher};
+    use sui::tx_context::{TxContext, sender};
+    use std::option::{Self, some, none};
+    use std::string::{Self, utf8};
+    use sui::object::{Self, UID};
+    use sui::event::emit;
+    use sui::transfer;
+
+    /// For when T does not belong to package in PublisherCap.
+    const ENotOwner: u64 = 0;
+
+    /// The display standard. Defines the way an object should be
+    /// displayed. New Display objects can only be created with a
+    /// PublisherCap, making sure that the rules are set by the owner
+    /// of the type.
+    ///
+    /// Each of the display properties should support patterns outside
+    /// of the system, making it simpler to customize Display based
+    /// on the property values of an Asset.
+    ///
+    /// Uses only String type for now; some fields may be replaced with
+    /// Url's but due to external-facing nature of the object, the
+    /// property names have a priority over types.
+    struct Display<phantom T: key> has key {
+        id: UID,
+        name: Option<String>,
```

Comment:
`VecMap<String, String>` might be one way to eliminate the options + enable extensibility.

Reply:
Or we could consider an alternative with dynamic fields... depending on which one is easier to implement.

Reply:
Both designs are definitely worth considering. The reason I suggest `VecMap` instead of dynamic fields is: 
- The storage cost of the dynamic fields design will be greater, possibly by a significant amount depending on the number of properties
- I expect that most folks reading the properties will want to retrieve all of them, which will be easier if they live in a single object

If a typical `Display` object has so many large properties that we start to bump up against the object, then dynamic fields start to look more appealing, but I'm guessing that typical display metadata will be pretty small.

Reply:
Done! Also added a hacky way to avoid unnecessary auth checks when a Display object is first created.